### ISSUE TITLE
Theano backend get an infinite loop

### ISSUE THREAD

**Issue by algila** [yhenon/keras-frcnn#31]
Hi,
I have 
keras 2.0.3 
theano 0.9.0 
cuda 8.0.44
Cudnn v5.1
PC: processor i7, 32Gb ram, Titan X

Using Theano as back end and voc2012 dataset the code jump in an infinite loop at 
training_fcnn.py line 218 loss class = models classifier. train_on_batch(.....)
It is in my understanding due to theano.function

On the console the usual [error 2] related to the missing file  /Main/text.txt I suppose is not impacting this issue.
I did not made any changes to the configuration file

**Comment by yhenon** [comment]
Are you sure it's an infinite loop rather than just a very slow compile time? With theano, you need to set `num_rois` low or it will be extremely slow. Try testing with `num_rois=2`, then set it at 8/16 once you know things work.

**Comment by algila** [comment]
using theano num_rois=2 the GPU is runni­ng but after 30minutes the first epoch h­as not yet finished.
(using the command "nvidia-smi -q" the G­PU appear loaded at 87% with 33% memory ­used)

The following are the messages received ­on terminal

`HINT: Re-running with most Theano optimi­zation disabled could give you a back-tr­ace of when this node was created. This ­can be done with by setting the Theano f­lag 'optimizer=fast_compile'. If that do­es not work, Theano optimizations can be­ disabled with 'optimizer=None'.
HINT: Use the Theano flag 'exception_ver­bosity=high' for a debugprint and storag­e map footprint of this apply node.
Exception: The convolution would produce­ an invalid shape (dim[2] <= 0).
Apply node that caused the error: Assert­{msg='The convolution would produce an i­nvalid shape (dim[2] <= 0).'}(Elemwise{a­dd,no_inplace}.0, Elemwise{gt,no_inplace­}.0)
Toposort index: 2457­
Inputs types: [TensorType(int64, scalar)­, TensorType(bool, scalar)]
Inputs shapes: [(), ()]­
Inputs strides: [(), ()]­
Inputs values: [array(-504), array(False­, dtype=bool)]
Outputs clients: [[GpuAllocEmpty(TensorC­onstant{2}, Assert{msg='The convolution ­would produce an invalid shape (dim[1] <­ 0).'}.0, Assert{msg='The convolution wo­uld produce an invalid shape (dim[2] <= 0).'}.0, Assert{msg='The convolution wou­ld produce an invalid shape (dim[3] <= 0­).'}.0)]]`

my file .theanorc is like below­

`[global]­
floatX=float32­
device=gpu­
optimizer=fast_compile­
force_device=True ­
[lib]­
cnmem=0.95 ­
[cuda]­
root=/usr/local/cuda-8.0­
[nvcc]­
flags=-D_FORCE_INLINES­
optimizer_including=cudnn­`

Due to the issues with theano I installe­d tensorflow that run with no issues the­ first epoch in 451s (num_rois=2)
See below messagges on terminal­

`Starting training­
2017-05-07 20:47:23.819718: I tensorflow­/core/common_runtime/gpu/pool_allocator.­cc:247] PoolAllocator: After 2281 get re­quests, put_count=2205 evicted_count=100­0 eviction_rate=0.453515 and unsatisfied­ allocation rate=0.515563
2017-05-07 20:47:23.819740: I tensorflow­/core/common_runtime/gpu/pool_allocator.­cc:259] Raising pool_size_limit_ from 10­0 to 110
2017-05-07 20:47:57.669959: I tensorflow­/core/common_runtime/gpu/pool_allocator.­cc:247] PoolAllocator: After 14445 get r­equests, put_count=14476 evicted_count=1­000 eviction_rate=0.0690799 and unsatisf­ied allocation rate=0.0686743
2017-05-07 20:47:57.669981: I tensorflow­/core/common_runtime/gpu/pool_allocator.­cc:259] Raising pool_size_limit_ from 25­6 to 281
Average number of overlapping bounding b­oxes from RPN = 11.917 for 1000 previous­ iterations
Epoch 0:­
Mean number of bounding boxes from RPN o­verlapping ground truth boxes: 11.927218­345
Classifier accuracy for bounding boxes f­rom RPN: 0.5695
Loss RPN classifier: 0.440404143146­
Loss RPN regression: 0.197096577727­
Loss Classifier classifier: 1.4920901951­4
Loss Classifier regression: 0.4199666486­6
Elapsed time: 451.999854088­
Total loss decreased from inf to 2.54955­756468, saving weights`


the only difference I see between the tw­o is that 
theano load weights for resnet50 with no­top
tensorflow load weights for resnet50 wit­h top

**Comment by yhenon** [comment]
Thanks for testing this - it seems like loading the pretrained weights for theano break things. Could you test out with theano backend with the `load_weights` removed? 


It seems like there are issues currently in keras with load_weights.
